components/FavoriteButton/FavoriteButton.js

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect, useState } from "react";
+import useLocalStorageState from "use-local-storage-state";
+import { MarkIcon, ColoredMarkIcon } from "../Icons/Icons";
+import { StyledMarkButton } from "./styles";
+
+export default function FavoriteButton({ article }) {
+  const [favorites, setFavorites] = useLocalStorageState("favorites", []);
+  const [isFavorite, setIsFavorite] = useState(false);
+
```

Comment:
A simple solution with the id could be:
if favorites.contains(id) {setIsFavorite(true)}

then the handleMarkFavorite has only to add or remove it from the favorites-state.

Reply:
Here it is the same. A component should be dump as can be. There are only two things, the favorite-button needs to know: is it favorite? And what to do when clicked.
Both you can pass via props (isFavorite, handleClick).
Please keep this in mind, because it is very important for development.